maven/WFLY-14934_Server_Plugin.adoc

```diff
@@ -0,0 +1,217 @@
+
+Today, in order to execute WildFly in an Openshift context, we are relying on WildFly S2I (Source-to-Image) builder and runtime images to build 
+and execute an WildFly application image.
+There is a strong coupling inside this image between the server version and the docker image that implies 
+releasing a new image for each new WildFly release. This approach is not flexible. We should be able to build 
+an application image for any WildFly release (even for a locally built SNAPSHOT) without having to wait for the 
+WildFly builder image to be built/deployed. 
+
+In addition, the WildFly server located inside the builder image, enforces a set of configuration API (a set of environment variables)
+and bash launch scripts. At startup time, a complex CLI script generation that impacts the server startup time is mandatorily executed.
+Having this level of configurability should be a user choice and not be enforced by the WildFly s2i image.
+
+To solve these pain points, we are removing in WFLY-14936 the server installation and its configuration aspects from the image.
+
+The WildFly S2I image becomes a generic image allowing the installation and execution of any provisioned WildFly server.
+
+Due to this removal, we need a way for customers to provision the WildFly server to be installed inside the new image 
+when building their application. This WildFly server will run the built application inside the image.
+
+To do so we are defining Maven tooling that handles the Galleon provisioning of the server, the application deployment and the configuration aspects.
+
+== Requirements
+
+The existing https://github.com/wildfly/wildfly-maven-plugin/[WildFly Maven plugin] has been designed to interact (mainly) with a running server instance. 
+We are here proposing to evolve the capabilities of this plugin for one to be able to build a fully configured server containing application deployment.
+ 
+This proposal adds a new kind of installation that is operated and controlled by the pom.xml file.  The pom.xml file contains all the information needed
+to control the server and application content and versioning.
+
+=== Requirements for cloud integration
+
+* The new ``package`` goal that this proposal adds to the Maven plugin is the goal to use during s2i build. 
+Other goals (e.g.: ``provisioning``, ``execute-commands``, ...) are not expected to be used.
+
+* The plugin can be used in a "cloud build" context. In particular, the plugin must be usable with:
+** WildFly s2i V2 builder image for s2i in binary and source (see: https://issues.redhat.com/browse/WFLY-14936[WFLY-14934])
+** JKube s2i and docker binary builds (see https://issues.redhat.com/browse/WFLY-14938[WFLY-14938])
+* The generated server can be installed and run successfully inside a WildFly runtime image v2 defined 
+   in https://issues.redhat.com/browse/WFLY-14936[WFLY-14936].
+
+The integration of this plugin for cloud integration will affect other projects:
```

Comment:
I suggest to add an explicit note that changes in mentioned projects are not a part of this RFE and that they can be improved by leveraging this RFE to make things finally clear. Preferably in/around "The integration of this plugin for cloud integration will affect other projects:" including " Eventually, other technologie such as BuildPacks will be able to leverate this plugin to build cloud images for WildFly.".